### DISCUSSION TITLE
Visual Studio 存在多版本工具链时，如何指定统一的MSVC ？

### DISCUSSION THREAD

**Discussion by AoyamaSun** [xmake-io/xmake#2324] [Q&A]
**现象描述：**
VS2019中安装了14.16版（VS2017）编译工具链，通过以下命令可以使用14.16版编译工程：
“xmake f --vs_toolset=14.16.27023 --vs_sdkver=10.0.18362.0 ”

但工程中存在依赖的packages时，编译packages还是用的高版本（14.29）编译器：
"D:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\**14.29.30133**\\bin\\HostX64\\x64\\cl.exe" -c /EHsc -nologo -std:c++17 -MDd -Isrc -Ithirdparty -Ithirdparty\fmt-6.1.2\include -Ithirdparty\sol2-3.2.2\include -Ithirdparty\asio-1.12.2\include -Ith...

通过修改目标package的xmake.lua，指定
        import("package.tools.xmake").install(package, configs, {cmake_generator = "Ninja"})
则xmake按照vs_toolset指定的版本（14.16）调用编译器。

**期望行为：**编译依赖package时按vs_toolset指定版本调用编译器。

**试用版本：** xmake v2.6.4+master.db869ffcd, A cross-platform build utility based on Lua

感谢xmake作者的付出。

**Comment by waruqi** [comment]
更新到 dev 试试，xmake update -s dev
